src/BlockNLPModels.jl

```diff
@@ -0,0 +1,213 @@
+module BlockNLPModels
+
+using NLPModels
+using LinearAlgebra
+using SparseArrays
+
+export AbstractBlockNLPModel
+export BlockNLPModel
+export DualizedNLPblockModel
+export FullSpaceModel
+export add_block, add_links, n_tot_con, get_blockmatrix, get_matrixblocks, get_RHSvector
+
+"""
+    AbstractBlockNLPModel
+Abstract supertype for the definition of NLP problems with a block structure.
+"""
+abstract type AbstractBlockNLPModel end
+
+"""
+    AbstractLinearLinkConstraint
+Abstract supertype for the definition of linear linking constraints
+"""
+abstract type AbstractLinearLinkConstraint end
+
+"""
+    AbstractNLLinkConstraint
+Abstract supertype for the definition of nonlinear linking constraints
+"""
+abstract type AbstractNLLinkConstraint end
+
+include("utils.jl")
+
+"""
+  BlockNLP_Counters()
```

Comment:
To follow the [Julia style guide](https://docs.julialang.org/en/v1/manual/style-guide/), I would rename this function `BlockNLPCounters` (we should avoid underscore when using CamelCase names)